ospfd/ospf_vty.c

```diff
@@ -12952,6 +12952,45 @@ DEFUN (clear_ip_ospf_interface,
 	return CMD_SUCCESS;
 }
 
+DEFPY_HIDDEN(ospf_lsa_refresh_timer, ospf_lsa_refresh_timer_cmd,
+	     "[no$no] ospf lsa-refresh [(120-1800)]$value",
+	     NO_STR
+	     "OSPF protocol\n"
+	     "OSPF lsa refresh timer\n"
+	     "timer value in seconds\n")
+{
+	VTY_DECLVAR_INSTANCE_CONTEXT(ospf, ospf)
+
+	if (no) {
+		ospf->lsa_refresh_timer = OSPF_LS_REFRESH_TIME;
+	} else {
+		ospf->lsa_refresh_timer = value;
+	}
+
+	return CMD_SUCCESS;
+}
+
+DEFPY_HIDDEN(ospf_maxage_delay_timer, ospf_maxage_delay_timer_cmd,
+	     "[no$no] ospf maxage-delay [(0-60)]$value",
+	     NO_STR
+	     "OSPF protocol\n"
+	     "OSPF lsa maxage delay timer\n"
+	     "timer value in seconds\n")
+{
+	VTY_DECLVAR_INSTANCE_CONTEXT(ospf, ospf)
+
+	if (no) {
+		ospf->maxage_delay = OSPF_LSA_MAXAGE_REMOVE_DELAY_DEFAULT;
+	} else {
+		ospf->maxage_delay = value;
+	}
+
+	OSPF_TIMER_ON(ospf->t_maxage, ospf_maxage_lsa_remover,
```

Comment:
Stopping an existing thread is not necessary before starting? Or is this a non-valid state?

Reply:
ya sorry I missed that, I thought OSPF_TIMER_ON internally took care of that, but updated in the latest patch.